Range of updates

JayManley opened the pull request:
Including;
•Save password option with encryption
•New player projectile effects(projectile movement is has been improved when using the CreatePlayerEffect scripting function)
•New bug reporting feature which gives players the opportunity to give us feedback and raise awareness to any bugs. (gods can view submitted bug reports by using /God Bugs)
•Added a local host option which only shows if the source project file is located in the application launch directory

JayManley committed:
Range of updates

Including;
•Save password option with encryption
•New player projectile effects(projectile movement is has been improved when using the CreatePlayerEffect scripting function)
•New bug reporting feature which gives players the opportunity to give us feedback and raise awareness to any bugs. (gods can view submitted bug reports by using /God Bugs)
•Added a local host option which only shows if the source project file is located in the application launch directory

JayManley commented:
I also made it so that gods can see invisible players in this update, I thought it made sense lol

clowenhg commented on modArray.bas:
```diff
@@ -350,6 +361,7 @@ Public Character As CharacterData
 Attribute Character.VB_VarUserMemId = 1073741840
 Public Macro(0 To 9) As MacroData
 Attribute Macro.VB_VarUserMemId = 1073741841
+Public Bug(1 To 500) As BugReportData
```
What happens at 500?

JayManley replied:
I’m aware of this but haven’t added in the error handling code yet

clowenhg commented on modMovement.bas:
```diff
@@ -129,7 +129,7 @@ Dim A As Long, B As Long, C As Long, D As Long
 
 Public Sub MoveProjectiles()
-Dim A As Long, B As Long, C As Long, D As Long
+Dim A As Long, B As Long, C As Long, D As Long, H As Double, X As Long, Y As Long, NewX As Long, NewY As Long
```
What is H?
At least for new variables, let's try to use useful names.

JayManley replied:
H was short for Hypotenuse. However these dimensions were eventually unused and therefore can be removed which I will also do...

clowenhg commented on modOdyssey.bas:
```diff
@@ -22,7 +24,11 @@ Sub Main()
     ServerDescription = "Odyssey Realms"
     CacheDirectory = App.Path + "\classic"
-    ServerIP = "libertyarchives.info"
+    If Exists("Odyssey.vbp") Then
```
Does this mean that with the vbp file in the directory you can only ever connect to localhost?

JayManley replied:
No, it means that if you have the vbp file then you have an extra option in the server list to connect to local host

clowenhg commented on modOdyssey.bas:
```diff
@@ -293,6 +299,11 @@ Sub MonsterDied(index As Long)
 End Sub
 
+Public Function GetOffsetDistance(SrcXO As Long, SrcYO As Long, DestXO As Long, DestYO As Long) As Long
+Dim A As Long, B As Long, C As Long, Dir As Long, XDist As Long, YDist As Long
```
Looks like none of these are used.

JayManley replied:
Yeah, these were unused. I was trying to get projectiles to create a straight line towards the target without using sourceX and sourceY

clowenhg commented on modOdyssey.bas:
```diff
@@ -373,7 +384,7 @@ Sub DisplayRepair()
             A = 0
             frmMain.lblMenu(56).Visible = False
-        ElseIf ExamineBit(Object(D).flags, 0) = 255 Or Object(D).SellPrice = 0 Then
+        ElseIf ExamineBit(Object(D).flags, 0) = 255 Then
```
Was this preventing repair on objects that should allow it?

clowenhg replied:
Or was SellPrice somehow identifying a completely broken item?

JayManley replied:
The object could not be repaired if sell price was set to 0 as the object repair cost was calculated from it. I changed it so that repair cost is now calculated using 500+.sellprice instead. If you think 500 gold as a minimum repair cost for all items is too much then it can be changed

JayManley replied:
2500* not 500. However 500 May have been a better option

clowenhg commented on modOdyssey.bas:
```diff
@@ -588,7 +599,7 @@ Function GetRepairCost(Slot As Integer) As Long
                 'If B > 0 Then B = B / 100
                 If Object(Character.Inv(Slot).Object).MaxDur * 10 > 0 Then
-                    C = Object(Character.Inv(Slot).Object).SellPrice - ((Character.Inv(Slot).value / (Object(Character.Inv(Slot).Object).MaxDur * 10)) * Object(Character.Inv(Slot).Object).SellPrice)
+                    C = Object(Character.Inv(Slot).Object).SellPrice + 2500 - ((Character.Inv(Slot).value / (Object(Character.Inv(Slot).Object).MaxDur * 10)) * (Object(Character.Inv(Slot).Object).SellPrice + 1))
```
Is there generally a change to how SellPrice is being used now?

clowenhg commented on modSecurity.bas:
```diff
@@ -204,6 +204,48 @@ Function StripNulls(ByVal sStr As String) As String
     StripNulls = Left$(sStr, lstrlen(sStr))
 End Function
 
+Public Sub SavePassword(Password As String)
```
This is not really secure.
Especially given that our community frequently sends their odyssey folders to each other, this presents an issue.
A short term solution would be to write the password out to a folder other than the odyssey one. Perhaps in the user's home directory or appdata.
Long term, we'll need some other solution.

JayManley committed:
Cleared unused code + Client Version update

•Cleared out the unused dimensions and function that @clowenhg pointed out. •Changed the client version to 3 in order to match the server version.

JayManley committed:
Sorry, these variables are required...

...

QBcrusher committed:
Merge branch 'develop' into Jay's-Updates
